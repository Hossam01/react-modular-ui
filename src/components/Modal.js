import React from 'react';

import AbstractComponent from './AbstractComponent';
import Portal from './Portal';

export default class Modal extends AbstractComponent {
  onClose() {
    if (typeof this.props.onClose === 'function') {
      this.props.onClose();
    }
  }

  onClickContent(event) {
    event.stopPropagation();
  }

  render() {
    return (
      <Portal show={this.props.show} portal={this.props.portal} portalKey={this.props.name}>
        <div className={this.getBlockClassName('modal')} onClick={() => this.onClose()}>
          <div className={this.getElementClassName('modal', 'content')} onClick={(event) => this.onClickContent(event)}>
            {this.props.children}
          </div>
        </div>
      </Portal>
    );
  }
}
