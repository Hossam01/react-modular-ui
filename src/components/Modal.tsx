import * as React from 'react';

import AbstractComponent from './AbstractComponent';
import Portal from './Portal';

export interface IModalProps {
  show?: boolean;
  portal?: JSX.Element[];
  name?: string;
  onClose?: () => void;
}

export default class Modal extends AbstractComponent<IModalProps, {}> {
  onClose() {
    if (typeof this.props.onClose === 'function') {
      this.props.onClose();
    }
  }

  onClickContent(event: React.MouseEvent<HTMLDivElement>) {
    event.stopPropagation();
  }

  render() {
    return (
      <Portal show={this.props.show} portal={this.props.portal} portalKey={this.props.name}>
        <div className={this.getBlockClassName('modal')} onClick={() => this.onClose()}>
          <div className={this.getElementClassName('modal', 'content')} onClick={(event) => this.onClickContent(event)}>
            {this.props.children}
          </div>
        </div>
      </Portal>
    );
  }
}
