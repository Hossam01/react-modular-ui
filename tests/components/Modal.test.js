import React from 'react';
import renderer from 'react-test-renderer';
import Modal from '../../src/components/Modal';

describe('Modal', () => {
  it('should render Modal', () => {
    const tree = renderer.create(
      <Modal>Something</Modal>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
