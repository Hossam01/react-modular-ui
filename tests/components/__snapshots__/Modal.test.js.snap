// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render Modal 1`] = `null`;
